web/html/src/components/table/SearchField.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from "react";
+
+type SearchFieldProps = {
+  /** Search criteria value */
+  criteria?: string;
+
+  /** Place holder value to display when nothing has been input */
+  placeholder: string;
+
+  /** function called when a search is performed.
+   * This is usually passed by the search panel parent component.
+   */
+  onSearch?: (arg0: string) => void;
+
+  /** filtering function */
+  filter?: (datum: any, criteria?: string) => boolean;
+
+  /** input field name */
+  name?: string;
+};
+
+/** Text input search field */
+export class SearchField extends React.Component<SearchFieldProps> {
```

Comment:
This is a workaround because I couldn't get the functional version to work with the way we use it as a prop for `Table.tsx` (and elsewhere):  

```ts
type TableProps = {
  /** the React Object that contains the filter search field */
  searchField?: React.ReactComponentElement<typeof SearchField>;
  // ...
}
```

We rely on the exact types of this lining up in the consuming component so the types do matter here.  
For the time being, I couldn't find a good fix for this as a functional component and wanted to move forward with the rest of the problematic files. If you have a good suggestion to this, of course I'm all ears.

Reply:
Fair enough :+1: